test/functional/lua/utility_functions_spec.lua

```diff
@@ -285,7 +289,15 @@ describe('lua stdlib', function()
   end)
 
   it('vim.pesc', function()
-    eq('foo%-bar', exec_lua([[return vim.pesc('foo-bar')]]))
-    eq('foo%%%-bar', exec_lua([[return vim.pesc(vim.pesc('foo-bar'))]]))
+    local pesc = function(s)
+      return exec_lua('return vim.pesc(...)', s)
+    end
+
+    eq('foo%-bar', pesc('foo-bar'))
+    eq('foo%%%-bar', pesc(pesc('foo-bar')))
+
+    -- type checked args
+    local path_pattern = iswin() and '[a-zA-Z]:[^:]+:%d+:' or '[^:]+:%d+:'
```

Comment:
@justinmk So that's it!  I couldn't have thought of this kind of workaround!  I thank you for your mentoring and I will do my best to submit a better PR. :)